include/nuttx/sched_note.h

```diff
@@ -99,6 +99,15 @@
   memset((s), 0, sizeof(struct note_filter_irq_s))
 #endif
 
+/* Note dump module tag definitions */
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_DUMP
+#  define NOTE_MODULE(a, b, c, d) \
+  ((unsigned)(a) | ((unsigned)(b) << 8) | ((unsigned)(c) << 16) | ((d) << 24))
```

Comment:
```suggestion
  ((a) | ((uint32_t)(b) << 8) | ((uint32_t)(c) << 16) | ((uint32_t)(d) << 24))
```

Reply:
@pkarashchenko  , Integer promotions

Reply:
yes. current variant has a bug when `(unsigned)(a)` is promoted (that is not needed), but `((d) << 24)` is not.
additionally in `void sched_note_vbprintf(uint32_t module, uint8_t event, FAR const char *fmt, va_list va)` the `module` is `uint32_t`, so there is not reason to use `unsigned` for promotion

Reply:
@pkarashchenko If `(a)`  is negative number，`((a) | *********) ` will promote wrong data.

Reply:
Actually if any of `a`, `b` or `c` are not byte size them result value will be wrong.
If you want to write reliable macro you need to add `& 0xff` to each argument.

Reply:
@ you are right